src/Tickets/Seating/Frontend.php

```diff
@@ -116,31 +116,33 @@ public function print_tickets_block( $html, $file, $name, $template, $context ):
 		}
 
 		$prices   = [];
-		$provider = Tickets::get_event_ticket_provider_object( $post_id );
-		foreach ( tribe_tickets()->where( 'event', $post_id )->get_ids( true ) as $ticket_id ) {
-			$ticket = $provider->get_ticket( $post_id, $ticket_id );
-			if ( ! $ticket ) {
-				continue;
-			}
-			$prices[] = $ticket->price;
-		}
+		$provider = Tickets::get_event_ticket_provider_object( $post_id ); // @todo - i see the below method can return false but we are never checking against it.
+
+		$tickets = array_filter(
+			array_map(
+				static function ( $ticket_id ) use ( $provider, $post_id ) {
+					$ticket = $provider->get_ticket( $post_id, $ticket_id );
+					if ( ! $ticket ) {
+						return false;
+					}
+
+					return $ticket;
+				},
+				tribe_tickets()->where( 'event', $post_id )->get_ids()
```

Comment:
`->get_ids()` will run an unbounded query. 
Whenever you're using the repository (not only for tickets), use a Generator that will run batches of queries.
That's the reason why the loop was using `foreach` originally; restore the `foreach` loop.